### PR TITLE
Mohsen/ fix: fix hyperlinks for derivlife in different languages

### PR DESCRIPTION
**Pull request by michio-rework**
Changes:

-   fix: hyperlinks for derivlife in different languages in several parts
of the app

## Type of change

-   [x] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

**Review by mohammad-rework — APPROVED** [review]

**Review by habib-deriv — DISMISSED** [review]

File: src/components/custom/other-platforms.js [hunk 1]
```diff
@@ -531,8 +531,9 @@ export const NavCompany = ({ onClick }) => (
         <CardLink
             icon={() => <img src={DerivLife} alt="" width="24" height="24" />}
             title={<Localize translate_text="Deriv life" />}
-            onClick={onClick}
-            to={deriv_life_url}
+            to={''}
+            type={'derivlife'}
```
Comment (sean-binary) [inline src/components/custom/other-platforms.js:535]:
```suggestion
            to=""
            type="derivlife"
```

File: src/components/elements/off-canvas-menu.js [hunk 2]
```diff
@@ -435,7 +435,9 @@ export const OffCanvasMenuWrapper = (props) => {
                             </SpanSvg>
                         </StyledLink>
                         <StyledLink
-                            to={deriv_life_url}
+                            to={''}
```
Comment (sean-binary) [inline src/components/elements/off-canvas-menu.js:438]:
```suggestion
                            to=""
```

File: src/components/layout/footer/main-links.js [hunk 3]
```diff
@@ -38,7 +38,13 @@ const MainLinksSection = ({ is_ppc, is_ppc_redirect }) => {
                             <Link to="/careers/">{<Localize translate_text="Careers" />}</Link>
                         </LinkWrapper>
                         <LinkWrapper>
-                            <Link to={deriv_life_url} target="_blank" rel="noopener noreferrer">
+                            <Link
+                                to={''}
```
Comment (sean-binary) [inline src/components/layout/footer/main-links.js:42]:
```suggestion
                                to=""
```

File: src/components/custom/other-platforms.js (hunk 1 above)
Comment (michio-rework) [reply src/components/custom/other-platforms.js]: suggested changes applied, thanks

File: src/components/elements/off-canvas-menu.js (hunk 2 above)
Comment (michio-rework) [reply src/components/elements/off-canvas-menu.js]: suggested changes applied, thanks

File: src/components/layout/footer/main-links.js (hunk 3 above)
Comment (michio-rework) [reply src/components/layout/footer/main-links.js]: suggested changes applied, thanks

File: src/components/custom/other-platforms.js [hunk 4]
```diff
@@ -531,8 +531,9 @@ export const NavCompany = ({ onClick }) => (
         <CardLink
             icon={() => <img src={DerivLife} alt="" width="24" height="24" />}
             title={<Localize translate_text="Deriv life" />}
-            onClick={onClick}
-            to={deriv_life_url}
+            to=""
```
Comment (prince-deriv) [inline src/components/custom/other-platforms.js:534]: I think this is optional and can be left an assigned.

Comment (michio-rework) [reply src/components/custom/other-platforms.js:534]: I though the same thing, but when I didn't provide the empty string in `to` prop my link would be something like : `https://derivlife.com/undefined`, this is caused by not providing default value for `to` param in `getURLFormat` in `src/components/localization/localized-link.js`, so Unfortunately I have to provide the empty string here.

**Review by habib-deriv — APPROVED** [review]

File: src/components/custom/other-platforms.js (hunk 4 above)
Comment (prince-deriv) [reply src/components/custom/other-platforms.js:534]: Alright. This will be handled once we refactor this file to typescript.

**Pull request merged**